### ISSUE TITLE
9.1.4.10 Inhalte brechen um - 320 CSS Pixel immer federführend?

### ISSUE THREAD

**Issue by sprungmarker** [BIK-BITV/BIK-Web-Test#261]
Bei etlichen Tests habe ich grobe Probleme in der Ansicht 1280 x 1024px (Viewport) festgestellt, die bei der 320 CSS Pixel Ansicht nicht auftreten.

Lt. Prüfschritt gehen in die Bewertung nur die Probleme der 320 CSS Pixel Ansicht ein. 

Aber gibt es nicht auch einen starken Anwendungsfall für die Ansicht 1280 x 1024px (Viewport)? Kann man die - bei groben Problemen - wirklich ausser Acht lassen und nicht bewerten?

**Comment by sweckenmann** [comment]
Hm, ich konnte dazu nur ein älteres issue von @detlevhfischer bei der Working Group zu dem Thema finden, siehe https://github.com/w3c/wcag/issues/648 - aber daraus sind keine Schlüsse zu ziehen?

**Comment by stefanFarnetani** [comment]
@sprungmarker Kannst du genauer schildern welche Art die Probleme sind? handelt es sich evtl. um einen bestimmten Typus von Problemen?

**Comment by sprungmarker** [comment]
@stefanFarnetani ja, im Grunde sind es bei 400% Zoom (1280x1024) immer Probleme, dass Inhalte nicht erreichbar sind, abgeschnitten (hauptsächliche Navigationsbereiche). Das ist ja keine Problematik, die in diesem Prüfschritt zu einem Abzug führt. Bei 320 CSS Pixel gibt es diese Probleme manchmal auch, aber manchmal auch gar nicht.

Generell ist es schon so, dass sich die Probleme bei 400% noch verschärfen, aber manchmal - und das kommt bei mir nicht so oft vor, gibt es einfach Probleme (Überlagerungen, Inhalte werden gescrollt, abgeschnitten etc.), die bei 320 CSS Pixel nicht vorhanden sind.

**Comment by MarcHaunschild** [comment]
Bei mir ist es auch so, wie @sprungmarker es beschreibt. Kann das so nicht genau benennen. Ich für meinen Teil dachte immer, es ist halt kein Zoom-Prüfschritt (das ist der [9.1.4.4 Text auf 200 % vergrößerbar](https://webtest.bitv-test.de/index.php?a=di&iid=261&s=n)) - daher teste ich mit verschiedenen Methoden eine Bildschirmbreite bei 320px. In der Regel, indem ich die Entwicklerwerkzeuge neben dem Viewport öffne, das html-Element auswähle, und den Viewport so weit zusammen schiebe, bis mir die Entwicklerwerkzeuge eine Breite von 320px anzeigen). Das ist schon Fummelei und muss für jede Seite neu gemacht werden, da Firefox nicht die Breite von Fenster und Entwicklerwerkzeuge für alle Tabs gleich einstellt.
So oder so gibt es allerhand Varianten (mit oder ohne Scrollbalken) und die Methode auf 400% zu zoomen bei 1280px breitem Fenster hat wieder ganz andere. Ich finde um die Vergleichbarkeit der Ergebnisse zu gewährleisten sollte hier ein ganz bestimmtes Verfahren festgelegt und einheitlich durchgeführt werden und wie gesagt: um zoomen geht es in dem Prüfschritt hier eigentlich nicht...

**Comment by stefanFarnetani** [comment]
Bin ebenfalls für eine einheitliche Messmethode. Also immer mit dem kleinen Bildschirm testen.
Beim Testen muss man ja oft eine Vereinfachung festlegen. Diese scheint sinnvoll zu sein.

Dann passt es wenn auch gleichzeitig die Änderung aus issue #259 mit umgesetzt werden, oder?

**Comment by sprungmarker** [comment]
Gut, soweit ich sehe ist alles geklärt - dann schliesse ich jetzt dieses Issue.

**Comment by cstrobbe** [comment]
Ich verstehe nicht, warum dieses Issue geschlossen wurde. Der Prüfschritt sagt immer noch, dass "Seiten-Inhalte [...] bei einer Browserfensterbreite von 320 CSS-Pixeln (bzw. bei einer Browserfensterbreite von 1280 CSS-Pixeln und 400% Zoomvergrößerung) so umbrechen [sollen]", obwohl beide Testmethoden oft zu unterschiedlichen Ergebnissen führen.
Aus meiner Sicht zählt aber nur die Prüfung bei einer Browserfensterbreite von 320 CSS-Pixeln, da der "Note" zum Erfolgskriterium nicht normativ ist. Die alternative Prüfung mit 1280px Viewport-Breite sollte meines Erachtens entfern werden, weil sie nicht auf normativen Text basiert und nicht zum gleichen Ergebnis führt als die Prüfung mit 320px Viewport-Breite.

**Comment by detlevhfischer** [comment]
Diese Anforderung ist ja insbesondere forget Sehbehinderte Nutzende geschaffen worden, die mit starker Vergrößerung arbeiten. Der BI BITV-Test soll ja nicht nur Konformitöät feststellen (da stimme ich zu, dass für die Bewertung die Überprüfung bei 320 Px Breite reicht), sondern Entwicklern wichtige Hinweise geben - hier gerade zu Problemen, die bei der Nutzung von fest positionierten Bereichen oben oder unten auftreten. Die Prüfung bei 1280 px mit 400% hat also einen wichtigen heuristischen Wert. Das sollte vielleicht in dem Abschnitt “Warum…” nich deutlicher werden.
Ich schlage vor, die Bewertung auf die Ansicht bei 320 px und die Prüfung mit der Nutzung der “mobilen Ansicht” in den Devtools durchzuführen und ausschließlich diese Ansicht zugrunde zu legen (hier werden noch beide Verfahren genannt), die alternative Prüfung aber drin zu lassen, damit uns wichtige Hinweise an die Entwickler nicht verloren gehen.

**Comment by MarcHaunschild** [comment]
Hmm - wenn man hier eh schon auf 400% prüft, wofür braucht man dann noch den zusätzlichen Prüfschritt mit 200%?
Macht für mich keinen Sinn.
Für mich war die Abgrenzung kleine Bildschirme hier, Zoom bei 200%.
Keine Ahnung warum nur 200%, aber so ist es halt festgesetzt.

**Comment by cstrobbe** [comment]
Noch zur Information: SC 1.4.4 (Resize text) wurde damals auf 200% beschränkt mit der Begründung, dass Personen, die mehr als 200% Vergrößerung brauchen, eine andere Technologie brauchen als den Browserzoom, nämlich eine richtige Bildschirmlupe (auf dem Betriebssystem). Der Browserzoom vergrößert ja nur die Webinhalte, und nicht die Browser-UI, die man ja auch lesen will.
Deswegen überzeugt das Argument mit 400% Zoom nicht: Wem dieses Niveau von Vergrößerung ist mit Browserzoom nicht geholfen.

**Comment by detlevhfischer** [comment]
@MarcHaunschild Bitte nicht 1.4.4 und 1.4.10 durcheinander werfen. Die Vergrößerbarkeit auf 200% berücksichtigt nicht, ob die Seite ggf. horizontal gescrollt werden muss. Das Inhalte bei starker Vergrößerung umbrechen, ohne das horizontal gescrollt werden muss, war tatsächlich der Haupt-Treiber für diese Anforderung (dass Inhalte responsiv brauchbar angezeigt werden, haben Frontend-Entwickler heutzutage denke ich klar auf dem Schirm). Deshalb halte ich die Prüfung über Vergrößerung des Desktops für praktisch relevant, selbst wenn Inhalte, die wegen des positionierter nur noch in schmalen Schlitzen erscheinen, normativ gesehen nicht zu einem FAIl führen sollten. Aber wir wollen ja Entwicklern nützliche Hinweise geben, und dafür ist die Prüfung bei 1280 / 400% sehr nützlich.

**Comment by MarcHaunschild** [comment]
> selbst wenn Inhalte, die wegen fest positionierter Bereiche nur noch in schmalen Schlitzen erscheinen

Das ändert sich ja dann mit der WCAG 2.2...

Danke für die Erklärung!

**Comment by johannesFischer84** [comment]
@detlevhfischer 
> Ich schlage vor, die **Bewertung** auf die Ansicht bei 320 px und die Prüfung mit der Nutzung der “mobilen Ansicht” in den Devtools durchzuführen und ausschließlich diese Ansicht zugrunde zu legen (hier werden noch beide Verfahren genannt), **die alternative Prüfung aber drin zu lassen**, damit uns wichtige Hinweise an die Entwickler nicht verloren gehen.

Detlev, das finde ich so in Ordnung. Der Abschnitt "Bewertung" kann sich auch nur auf die 320 px beziehen. Die 400% Alternativ-Prüfung sollte zumindest in der Anleitung bleiben. Denn zwar ist das in den WCAG nur in einer NOTE erwähnt, jedoch tauchen die 400 % häufiger im Understanding-Dokument von 1.4.10 auf und auch die Prüfprozeduren in den Techniques geben zur Prüfung den 400%-Zoom an. Daher sollte das auf keinen Fall komplett aus der Anleitung entfernt werden, wenn sich der Test nah an die WCAG halten will.
